framework/include/utils/NestedSolve.h

```diff
@@ -70,6 +70,8 @@ class NestedSolve
 
   void setRelativeTolerance(Real rel) { _relative_tolerance_square = rel * rel; }
   void setAbsoluteTolerance(Real abs) { _absolute_tolerance_square = abs * abs; }
+  void setMinIterations(unsigned int min_iter) { _min_iterations = min_iter; }
+  void setMaxIterations(unsigned int max_iter) { _max_iterations = max_iter; }
```

Comment:
Those are not required as the members are public. The other two setters only exist because we store teh squares of these values.

Reply:
@dschwen Without these two methods, how can the user overwrite the default min and max iteration numbers and pass them to the subConcentration material?

Reply:
by simply doing 

```
nestedSolve._min_iterations = min_iter;
nestedSolve._max_iterations = max_iter;
```

in the host material

Reply:
Or - even better - by taking advantage of the validParams that NestedSolve provides, and using the NestedSolve constructor that takes an input parameters object. (and using params.set to change the default for the user - see the last email I sent you)

Reply:
@dschwen I agree that `nestedSolve._min_iterations = min_iter;` is much simpler. However, despite the many hints you've given me, I am still a bit unclear about the `params.set` approach. In `params.set<unsigned int> ("min_iterations") = some_value;`, that `some_value` is user-supplied in the host material, but also needs to be in the NestedSolve.C to override the default. You mentioned that NestedSolve does not have access to the material property interface. Then how do we pass that `some_value` from the host material to NestedSolve?

Reply:
`some_value` is _not_ user supplied. It is the default that your class can set that the user can override in the input file. Nested Solve has a [constructor](https://github.com/idaholab/moose/blob/next/framework/include/utils/NestedSolve.h#L28) that you can pass the input parameters of your class into.

Reply:
@dschwen Okay. Since we will set the default of min iteration in my material class, I am assuming that we do not need the `minIterationsDefault()` function in the NestedSolve anymore. In that case, what should we use to initialize the `_min_iterations()` in the NestedSolve default constructor? (It is currently initialized as `_min_iterations(minIterationsDefault())`).